src/system_database.ts

```diff
@@ -565,28 +575,102 @@ export class PostgresSystemDatabase implements SystemDatabase {
       throw new DBOSNonExistentWorkflowError(`Workflow ${originalWorkflowID} does not exist`);
     }
 
-    const workflowStatusInternal: WorkflowStatusInternal = {
-      ...workflowStatus,
-      workflowUUID: forkedWorkflowId,
-      status: StatusString.ENQUEUED,
-      workflowName: workflowStatus.workflowName,
-      workflowClassName: workflowStatus.workflowClassName,
-      workflowConfigName: workflowStatus.workflowConfigName,
-      queueName: INTERNAL_QUEUE_NAME,
-      authenticatedUser: workflowStatus.authenticatedUser,
-      authenticatedRoles: workflowStatus.authenticatedRoles,
-      request: workflowStatus.request,
-      executorId: globalParams.executorID,
-      applicationVersion: workflowStatus.applicationVersion,
-      applicationID: workflowStatus.applicationID,
-      createdAt: Date.now(),
-    };
+    const client = await this.pool.connect();
 
-    await this.initWorkflowStatus(workflowStatusInternal, inputs);
+    try {
+      await client.query('BEGIN');
 
-    await this.enqueueWorkflow(forkedWorkflowId, INTERNAL_QUEUE_NAME);
+      const query = `
+        INSERT INTO ${DBOSExecutor.systemDBSchemaName}.workflow_status (
+        workflow_uuid,
+        status,
+        name,
+        class_name,
+        config_name,
+        queue_name,
+        authenticated_user,
+        assumed_role,
+        authenticated_roles,
+        request,
+        output,
+        executor_id,
+        application_version,
+        application_id,
+        created_at,
+        recovery_attempts,
+        updated_at
+      ) VALUES($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17)
+        `;
+
+      await client.query(query, [
+        forkedWorkflowId,
+        StatusString.ENQUEUED,
+        workflowStatus.workflowName,
+        workflowStatus.workflowClassName,
+        workflowStatus.workflowConfigName,
+        INTERNAL_QUEUE_NAME,
+        workflowStatus.authenticatedUser,
+        workflowStatus.assumedRole,
+        DBOSJSON.stringify(workflowStatus.authenticatedRoles),
+        DBOSJSON.stringify(workflowStatus.request),
+        null,
+        workflowStatus.executorId,
```

Comment:
This should definitely be null, as there's no reason the forked workflow should run on the same executor as the original.